beluga_amcl/src/ndt_amcl_node_3d.cpp

```diff
@@ -245,8 +255,14 @@ beluga::NDTSensorModel<NDTMapRepresentation> NdtAmclNode3D::get_sensor_model() c
   const auto map_path = get_parameter("map_path").as_string();
   RCLCPP_INFO(get_logger(), "Loading map from %s.", map_path.c_str());
 
-  return beluga::NDTSensorModel<NDTMapRepresentation>{
-      params, beluga::io::load_from_hdf5<NDTMapRepresentation>(get_parameter("map_path").as_string())};
+  // Load the map from hdf5 file
+  const auto map = beluga::io::load_from_hdf5<NDTMapRepresentation>(get_parameter("map_path").as_string());
+
+  // Publish markers for map visualization
+  beluga_ros::msg::MarkerArray obstacle_markers = beluga_ros::assign_obstacle_map(map);
+  map_visualization_pub_->publish(obstacle_markers);
+
+  return beluga::NDTSensorModel<NDTMapRepresentation>{params, map};
```

Comment:
@Fernando-Sanz meta: I understand why this was added here but it is not a suitable place for publishing messages. Consider decoupling sensor model instantiation from map loading so you can publish these markers on node activate.

Reply:
this is called from `make_particle_filter()`, which is called on `on_activate()`.
We  could first save the loaded sparse grid and then separately publish on `on_activate()` but it's not going to make any real difference

Reply:
+1 to separate publishing